src/app/waste-water/explore-opportunities/explore-opportunities-form/explore-opportunities-form.component.ts

```diff
@@ -10,14 +10,26 @@ import { WasteWaterService } from '../../waste-water.service';
 })
 export class ExploreOpportunitiesFormComponent implements OnInit {
 
+  modificationExists: boolean;
   modification: WasteWaterData;
   selectedModificationIdSub: Subscription;
   constructor(private wasteWaterService: WasteWaterService) { }
 
   ngOnInit(): void {
     this.selectedModificationIdSub = this.wasteWaterService.selectedModificationId.subscribe(val => {
-      this.modification = this.wasteWaterService.getModificationFromId();
+      if(val){
+        this.modificationExists = true;
+        this.modification = this.wasteWaterService.getModificationFromId();
+      } else {
+        this.modificationExists = false;
+      }
+      
     });
+
+    if(!this.modification.name){
```

Comment:
Might try running with this if block removed. You're checking above before setting a modification, and an *ngIf="modification" on the formgroup block in the html can be added to prevent breaking in the event there is no modificaiton.